core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -269,12 +269,6 @@ public Builder withHeaders(Map<String, String> headers) {
       return this;
     }
 
-    public Builder withBearerAuth(String token) {
-      Preconditions.checkNotNull(token, "Invalid auth token: null");
-      baseHeaders.put(HttpHeaders.AUTHORIZATION, asBearer(token));
```

Comment:
nit: seems like we can get by without a separate method to format the bearer token?

Reply:
We could also use the static value from RESTCatalogProperties as opposed to hard coded 'Bearer'

Reply:
Yeah, I'll remove the unnecessary method as well. That should fix both.